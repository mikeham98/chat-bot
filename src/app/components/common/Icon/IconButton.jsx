import React from 'react';
import styles from '../../../../themes/components/icon.scss';
import Icon from "./index";

const IconButton = ({onClick, type, icon, style, color, hoverColor, height, width, disabled}) => {
    return (
        <button
            type={type}
            onClick={onClick}
            disabled={disabled}
            style={style}
            className={styles.iconButton}
        >
            <Icon
                IconComponent={icon}
                color={color}
                hoverColor={hoverColor}
                height={height}
                width={width}
            />
        </button>
    );
};

export default React.memo(IconButton);