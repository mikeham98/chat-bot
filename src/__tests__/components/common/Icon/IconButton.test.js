import React from 'react';
import {shallow} from 'enzyme';
import IconButton from '../../../../app/components/common/Icon/IconButton';

describe('IconButton', () => {
    let shallowedWrapper;
    let props = {};
    const wrapper = () => {
        if (!shallowedWrapper) {
            shallowedWrapper = shallow(<IconButton {...props}/>);
        }
        return shallowedWrapper;
    };

    beforeEach(() => {
        props = {
            onClick: jest.fn(),
            type: 'text',
            icon: 'I am an icon',
            style: {
                margin: '10px',
                padding: '10px'
            },
            color: 'red',
            hoverColor: 'blue',
            height: '20px',
            width: '20px',
            disabled: true
        };
        shallowedWrapper = undefined;
    });

    describe('render', () => {
        it('should return a button with an Icon component as a child', () => {
            expect(wrapper()).toMatchSnapshot();
        });
    });
    describe('functionality', () => {
        describe('props', () => {
            describe('onClick', () => {
                it('should be called onClick of button', () => {
                    expect(props.onClick).not.toHaveBeenCalled();
                    wrapper().find('button').first().props().onClick();
                    expect(props.onClick).toHaveBeenCalledTimes(1);
                });
            });
        });
    });
});