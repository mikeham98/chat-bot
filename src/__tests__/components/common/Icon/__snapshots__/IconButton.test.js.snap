// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton render should return a button with an Icon component as a child 1`] = `
<button
  className="iconButton"
  disabled={true}
  onClick={[MockFunction]}
  style={
    Object {
      "margin": "10px",
      "padding": "10px",
    }
  }
  type="text"
>
  <Icon
    IconComponent="I am an icon"
    color="red"
    height="20px"
    hoverColor="blue"
    width="20px"
  />
</button>
`;
